docs/docs/tutorial/part-4/index.mdx

```diff
@@ -0,0 +1,918 @@
+
+**A closer look at the configuration options:**
+
+When your site builds, `gatsby-source-filesystem` adds all the files in folder specified by the `path` option to the data layer.
+
+The `name` option in the configuration object gets set to the `sourceInstanceName` field for each file. This comes in handy when you want to source files from multiple folders. By giving each folder a different `name` option, you can build GraphQL queries that filter down to only a particular folder.
+
+</Announcement>
+
+2. Restart your local development server to make sure it picks up the configuration changes and adds your files to the data layer.
+
+3. In GraphiQL, try exploring the different fields within `allFile` to see what sorts of data you get back. Then build a query using the `allFile` field to get the name of all the files in your blog folder:
+
+```graphql
+query MyQuery {
+  allFile {
+    nodes {
+      name
+    }
+  }
+}
+```
+
+4. Run the query in GraphiQL. Your response in the Result Window should look something like the object below:
+
+```json
+{
+  "data": {
+    "allFile": {
+      "nodes": [
+        {
+          "name": "my-first-post"
+        },
+        {
+          "name": "another-post"
+        },
+        {
+          "name": "yet-another-post"
+        }
+      ]
+    }
+  },
+  "extensions": {}
+}
+```
+
+### Task: Use a page query to pull the list of post filenames into your blog page
+
+Now that you've built a GraphQL query that returns a list of your post filenames, it's time to render that data in your blog page!
+
+Using GraphQL queries in page components uses a slightly different syntax from queries in building-block components. In page components, you use **page queries**.
+
+<Collapsible
+  summary={<h4>Key Gatsby Concept: Pulling data into page components with page queries</h4>}
+>
+
+The process for making a query in a page component looks slightly different from `useStaticQuery`:
+
+1. Import the `graphql` tag from the Gatsby package.
+2. Export a variable that stores a templated string with the GraphQL query you want to run.
+    * When your site gets built, Gatsby will run your page query and pass the resulting data into your page component as a prop called `data`.
+    * Your page query needs to be defined outside of your page component. (With `useStaticQuery`, your query was defined inside your component.)
+3. Use the `data` prop in your page component, as needed. You can use the JavaScript dot operator (`.`) to choose fields off of the `data` prop.
+
+Here's a small example to show what this process looks like in practice:
+
+```javascript
+import * as React from 'react'
+
+// Step 1: Import the graphql tag
+import { graphql } from 'gatsby'
+
+const HomePage = ({ data }) => {
+  return (
+    <p>
+      { /* Step 3: Use the data in your component*/ }
+      { data.site.siteMetadata.description }
+    </p>
+  )
+}
+
+// Step 2: Export a page query
+export const query = graphql`
```

Comment:
No strong opinion on this so your choice:

The name of the variable (in this case `query`) can be any name so it mustn't be `query`. In case you wanted to mention this

Reply:
Good idea, I'll add a note about that in the follow-up PR!